src/TaskManager/Runner/Program.cs

```diff
@@ -0,0 +1,176 @@
+                await Task.Delay(100).ConfigureAwait(false);
+            }
+            Console.CancelKeyPress += (sender, eventArgs) =>
+                    {
+                        eventArgs.Cancel = true;
+                        exitEvent.Set();
+                    };
+
+            // await Task.Run(() =>
+            // {
+            //     var message = GenerateDispatchEvent(argoBaseUri, wmConfig.Value);
+            //     logger.LogInformation($"Queuing new job with correlation ID={message.CorrelationId}.");
+            //     publisher.Publish(messagingKeys.TaskDispatchRequest, message);
+            // }).ConfigureAwait(false);
+
+            exitEvent.WaitOne();
+            logger.LogInformation("Stopping Task Manager...");
+
+            await host.StopAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        private static Message GenerateDispatchEvent(string argBaseUri, WorkflowManagerOptions wmConfig)
+        {
+            var correlationId = Guid.NewGuid().ToString();
+            var message = new JsonMessage<TaskDispatchEvent>(new TaskDispatchEvent
+            {
+                WorkflowId = Guid.NewGuid().ToString(),
+                TaskId = Guid.NewGuid().ToString(),
+                ExecutionId = Guid.NewGuid().ToString(),
+                CorrelationId = correlationId,
+                TaskPluginType = "argo",
+            }, applicationId: "TaskManagerRunner", correlationId: correlationId, deliveryTag: "1");
+            message.Body.TaskPluginArguments.Add(Keys.BaseUrl, argBaseUri);
+            message.Body.TaskPluginArguments.Add(Keys.WorkflowTemplateName, "list-input-artifacts-template");
+            message.Body.TaskPluginArguments.Add(Keys.WorkflowTemplateEntrypoint, "s3-artifacts-template");
+            message.Body.TaskPluginArguments.Add(Keys.MessagingEnddpoint, @$"{wmConfig.Messaging.PublisherSettings["endpoint"]}/{wmConfig.Messaging.PublisherSettings["virtualHost"]}");
+            message.Body.TaskPluginArguments.Add(Keys.MessagingUsername, wmConfig.Messaging.PublisherSettings["username"]);
+            message.Body.TaskPluginArguments.Add(Keys.MessagingPassword, wmConfig.Messaging.PublisherSettings["password"]);
+            message.Body.TaskPluginArguments.Add(Keys.MessagingExchange, wmConfig.Messaging.PublisherSettings["exchange"]);
+            message.Body.TaskPluginArguments.Add(Keys.MessagingTopic, wmConfig.Messaging.Topics.TaskCallbackRequest);
+            message.Body.Inputs.Add(new Messaging.Common.Storage
+            {
+                Name = "input-dicom",
+                Endpoint = wmConfig.Storage.Settings["endpoint"],
+                Bucket = wmConfig.Storage.Settings["bucket"],
+                SecuredConnection = Convert.ToBoolean(wmConfig.Storage.Settings["securedConnection"]),
+                RelativeRootPath = "/e08b7d7d-f30c-4f31-87d5-8ce5049aa956/dcm"

```

Comment:
@mocsharp this seems oddly specific

Reply:
The `Runner` project is used to test the Task Manager independently. It won't be used for production.